segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/BucketUpdate.java

```diff
@@ -0,0 +1,200 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.server.tables;
+
+import io.pravega.common.util.ArrayView;
+import io.pravega.common.util.HashedArray;
+import io.pravega.segmentstore.server.tables.hashing.KeyHash;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import javax.annotation.concurrent.NotThreadSafe;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.val;
+
+/**
+ * Represents an update to a Table Bucket.
+ */
+@RequiredArgsConstructor
+@NotThreadSafe
+class BucketUpdate {
```

Comment:
Given the way these are passed around. I would feel better if this were a builder and it could return the final version of the object.